{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/dabardonov/hse_hw2_chip/blob/main/ChIP_seq.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "lrtRbP3fAvj8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8eb1a0a7-e430-4955-9164-a2e446ab8246"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "📦 Installing...\n",
            "📌 Adjusting configuration...\n",
            "🩹 Patching environment...\n",
            "⏲ Done in 0:00:42\n",
            "🔁 Restarting kernel...\n"
          ]
        }
      ],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XnbIt77YBOT9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "36dcdc5f-6aad-4b23-cbea-72f64f63d8c7"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - seqtk\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
            "    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge\n",
            "    conda-4.11.0               |   py37h89c1867_1         983 KB  conda-forge\n",
            "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         1.3 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_1\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.11.0-py37h89c1867_1\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00,  6.27it/s]                \n",
            "seqtk-1.3            | 38 KB     | : 100% 1.0/1 [00:00<00:00, 25.80it/s]\n",
            "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00, 18.66it/s]\n",
            "conda-4.11.0         | 983 KB    | : 100% 1.0/1 [00:00<00:00,  3.33it/s]\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "62978ce7-736b-49dd-9900-cb75b1876379"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00,  6.39it/s]                \n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 29.65it/s]\n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:13<00:00, 13.73s/it]             \n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "3192fcae-cfe6-46c2-f7f2-3a2c88565462"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.1 MB 8.5 MB/s \n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 5.5 MB 27.8 MB/s \n",
            "\u001b[?25hCollecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.5 MB 19.6 MB/s \n",
            "\u001b[?25hCollecting matplotlib\n",
            "  Downloading matplotlib-3.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.2 MB 46.8 MB/s \n",
            "\u001b[?25hCollecting pandas\n",
            "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.3 MB 44.2 MB/s \n",
            "\u001b[?25hCollecting numpy\n",
            "  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
            "Collecting scipy\n",
            "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 38.1 MB 1.2 MB/s \n",
            "\u001b[?25hCollecting seaborn\n",
            "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
            "\u001b[K     |████████████████████████████████| 292 kB 64.9 MB/s \n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.3.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 49.8 MB/s \n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.0.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.3 MB 38.3 MB/s \n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[K     |████████████████████████████████| 247 kB 63.0 MB/s \n",
            "\u001b[?25hCollecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 7.5 MB/s \n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.29.1-py3-none-any.whl (895 kB)\n",
            "\u001b[K     |████████████████████████████████| 895 kB 26.4 MB/s \n",
            "\u001b[?25hCollecting packaging>=20.0\n",
            "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "\u001b[K     |████████████████████████████████| 40 kB 5.9 MB/s \n",
            "\u001b[?25hRequirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Collecting pytz>=2017.3\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "\u001b[K     |████████████████████████████████| 503 kB 54.1 MB/s \n",
            "\u001b[?25hCollecting pysam\n",
            "  Downloading pysam-0.18.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 14.9 MB 21.2 MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=fd39d94a0ea09960d5af525439a1d4ad40223425b2ed7538e0480c37b685d745\n",
            "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
            "  Building wheel for macs2 (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790967 sha256=b3588c6e1e34c1583850734612b5d924d990693af0cf21c51ee7c7b07021bb51\n",
            "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654286 sha256=4cd373d402caca5832411a613357202ec0a0647b58e465a15ea71e7156b88170\n",
            "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
            "Successfully built intervene macs2 pybedtools\n",
            "Installing collected packages: pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
            "Successfully installed cycler-0.11.0 fonttools-4.29.1 intervene-0.6.5 kiwisolver-1.3.2 macs2-2.2.7.1 matplotlib-3.5.1 numpy-1.21.5 packaging-21.3 pandas-1.3.5 pillow-9.0.1 pybedtools-0.9.0 pyparsing-3.0.7 pysam-0.18.0 python-dateutil-2.8.2 pytz-2021.3 scipy-1.7.3 seaborn-0.11.2\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numpy",
                  "pyparsing"
                ]
              }
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "7f4b4358-ff23-4635-c8c3-0f9901dedca9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 1s (499 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155332 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "5c96f4d7-ad53-49a8-fb03-f51b976064a7"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 9,743 kB of archives.\n",
            "After this operation, 46.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 9,743 kB in 2s (4,376 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155383 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.4.1-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up bowtie2 (2.3.4.1-1) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "29cbb322-1424-43e3-e76b-48aaf93ad9b4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 18:44:33--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  2.56MB/s    in 3.8s    \n",
            "\n",
            "2022-03-09 18:44:38 (2.56 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "8f44a206-9d0d-4f11-c289-a993e37bbc38"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 18:45:39--  https://www.encodeproject.org/files/ENCFF195ZFO/@@download/ENCFF195ZFO.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/04a0a840-ff6d-41b1-8ec7-a11d97fe6627/ENCFF195ZFO.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF195ZFO.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=VviryIpN21L2xUlE6NbsAEiyP8U%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981139 [following]\n",
            "--2022-03-09 18:45:39--  https://encode-public.s3.amazonaws.com/2017/10/13/04a0a840-ff6d-41b1-8ec7-a11d97fe6627/ENCFF195ZFO.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF195ZFO.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=VviryIpN21L2xUlE6NbsAEiyP8U%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981139\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.144.74\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.144.74|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3381896063 (3.1G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF195ZFO.fastq.gz’\n",
            "\n",
            "ENCFF195ZFO.fastq.g 100%[===================>]   3.15G  47.2MB/s    in 78s     \n",
            "\n",
            "2022-03-09 18:46:58 (41.1 MB/s) - ‘ENCFF195ZFO.fastq.gz’ saved [3381896063/3381896063]\n",
            "\n",
            "--2022-03-09 18:46:58--  https://www.encodeproject.org/files/ENCFF734JYK/@@download/ENCFF734JYK.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/415f0016-d8c2-4bd7-a437-7b9baa51c2af/ENCFF734JYK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF734JYK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=EZE9FZQPgVq0B3ouvhlqDRVchxA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981218 [following]\n",
            "--2022-03-09 18:46:58--  https://encode-public.s3.amazonaws.com/2017/10/13/415f0016-d8c2-4bd7-a437-7b9baa51c2af/ENCFF734JYK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF734JYK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=EZE9FZQPgVq0B3ouvhlqDRVchxA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981218\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.130.193\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.130.193|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2723479828 (2.5G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF734JYK.fastq.gz’\n",
            "\n",
            "ENCFF734JYK.fastq.g 100%[===================>]   2.54G  43.5MB/s    in 62s     \n",
            "\n",
            "2022-03-09 18:48:00 (42.1 MB/s) - ‘ENCFF734JYK.fastq.gz’ saved [2723479828/2723479828]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF195ZFO/@@download/ENCFF195ZFO.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF734JYK/@@download/ENCFF734JYK.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF151SVH/@@download/ENCFF151SVH.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NxVWhYSbfshg",
        "outputId": "d1049dcf-2705-4e6c-b400-0aee458bde5f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 18:48:00--  https://www.encodeproject.org/files/ENCFF151SVH/@@download/ENCFF151SVH.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/bffb72a4-28e5-4f77-98b9-358feb1fd186/ENCFF151SVH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151SVH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=WPaQvEZNb3p99MhxdHofm0WfCpo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981280 [following]\n",
            "--2022-03-09 18:48:00--  https://encode-public.s3.amazonaws.com/2017/10/13/bffb72a4-28e5-4f77-98b9-358feb1fd186/ENCFF151SVH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151SVH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUFKHKUKT&Signature=WPaQvEZNb3p99MhxdHofm0WfCpo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHMaCXVzLXdlc3QtMiJGMEQCIHWw2%2Bfz5TrNkpeD%2BuOu3pR0ftABjCcxvhWlsm1nx8XhAiBNQ%2F8w5uUIO%2F5bk5fvqxLnMJPB8B2aB0Z1GdTaXBUP%2BSqDBAjc%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMdxmXybuYtTjmMugNKtcD8Y2a5%2FhSBHcign2ozY6yb4gRlAySfBOcZ6uEevrl%2B37M%2Fmc2e3weg5pOhxS%2FW%2FivQcgVljP61%2BILT%2BohvUP5DIwIueQomaXJm%2FBjMQdhd5DhqXg6ZlAEWfo%2Fw5gCsb7svTMr8bJtHTpHV3ttTm%2BaFqVf1GRT8IZp5KgYoUOsCNDG7ajwrlXNubaAZtB74quJ5FyYELGcTaxFDAKNy3jqg7RU8Yf%2BurSBlWC65MclvskE1QTeBREqalEBO92%2BkVTcl6pNcMLQ%2BZ%2FyTmVIvg9ZjHMACTDY6pLdCYD2ZUWLrmypC4pQu2fw8kq3d9zeT8EuMPYYAyKG%2FGNUoNnJxBPxZUNjg6uLEGJ6j7Tdh8BZmR9HzfsPCAQXURKHb8aAOPmA7b1GowudXWLHglIbNa%2BEmklDlC%2FZZuBhsUS%2FZgGHVG3LeE4K54H6FJVVs1HQywIAFgcVswIJjtDo9Mn2hYUQXZ8oRkYcMkeJ6GIWlnjr0d81vbrS%2B1vRvUYkMPra%2BKuKWTSloYv0f%2FG6DBvuUeNIQZu2BCY%2BLNv5NaNy0D7L1aqLy1Fx3HFvka%2F%2BXzruu5C2Ey4%2BArwvu%2FYNTRjhQNowRERlpbl1yEL814duY63UjiQK9OVBoO6VMNDho5EGOqYB24rH0%2BYeapvZwXTJCQSraF3ZYaTjVg%2BvTX4xO0w0wGEQskuwIGXRjrrUup8oxHoOHQAYPmHgjgq4V77%2Fj5MJDsxb8TNKdZ%2BV4jznUYqgqcPPgFHM5Fp94cJGGD3MaThP6hIqDxWGJtVo6DMj9%2BPMRcc7wmcI8RUqXyFQGP3Iu1w4YmzipkVdZ2g1DmWMKcbiOHqbcm4C1FKLoGveqeiXCsi55hxDrg%3D%3D&Expires=1646981280\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.248.194\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.248.194|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1974036260 (1.8G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF151SVH.fastq.gz’\n",
            "\n",
            "ENCFF151SVH.fastq.g 100%[===================>]   1.84G  27.5MB/s    in 58s     \n",
            "\n",
            "2022-03-09 18:48:58 (32.6 MB/s) - ‘ENCFF151SVH.fastq.gz’ saved [1974036260/1974036260]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "!gzip -d /content/ENCFF195ZFO.fastq.gz /content/ENCFF734JYK.fastq.gz /content/ENCFF151SVH.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5XM6jBcuGjMV"
      },
      "outputs": [],
      "source": [
        "!./FastQC/fastqc /content/ENCFF195ZFO.fastq\n",
        "!./FastQC/fastqc /content/ENCFF734JYK.fastq\n",
        "!./FastQC/fastqc /content/ENCFF151SVH.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "Подрезать чтения не нужно"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "N4JIGZOsGiY2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e0c3cac6-24a4-41bb-d2ea-e85eee26d66e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 18:56:45--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29295890 (28M) [application/x-gzip]\n",
            "Saving to: ‘chr14.fa.gz’\n",
            "\n",
            "chr14.fa.gz         100%[===================>]  27.94M  58.1MB/s    in 0.5s    \n",
            "\n",
            "2022-03-09 18:56:46 (58.1 MB/s) - ‘chr14.fa.gz’ saved [29295890/29295890]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
        "!gzip -d chr14.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "bhraQ6NdGiPN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d97e8e64-0c83-40da-c508-aa8a692c77a4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr14.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.05681e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 90568149 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:44\n",
            "Returning block of 90568150 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:03\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.05681e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 90568149 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:44\n",
            "Returning block of 90568150 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:03\n",
            "CPU times: user 994 ms, sys: 132 ms, total: 1.13 s\n",
            "Wall time: 2min 6s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr14.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9e608fc4-7066-44c0-f49c-37b47d8cabf4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "109131346 reads; of these:\n",
            "  109131346 (100.00%) were unpaired; of these:\n",
            "    94077115 (86.21%) aligned 0 times\n",
            "    4339598 (3.98%) aligned exactly 1 time\n",
            "    10714633 (9.82%) aligned >1 times\n",
            "13.79% overall alignment rate\n",
            "CPU times: user 26.2 s, sys: 3 s, total: 29.2 s\n",
            "Wall time: 1h 22min 40s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF195ZFO.fastq \\\n",
        "          -S bowtie2_res/res_ZFO.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9b337ebe-ada7-498a-b244-734f9da3fcb8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "88120815 reads; of these:\n",
            "  88120815 (100.00%) were unpaired; of these:\n",
            "    76904496 (87.27%) aligned 0 times\n",
            "    3340751 (3.79%) aligned exactly 1 time\n",
            "    7875568 (8.94%) aligned >1 times\n",
            "12.73% overall alignment rate\n",
            "CPU times: user 32.1 s, sys: 3.8 s, total: 35.9 s\n",
            "Wall time: 1h 7min 20s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF734JYK.fastq \\\n",
        "          -S bowtie2_res/res_JYK.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF151SVH.fastq \\\n",
        "          -S bowtie2_res/res_SVH.sam"
      ],
      "metadata": {
        "id": "Sc4zGUdUp9eP",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0f4b7117-4314-409d-dec9-d2481015145f"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "64750538 reads; of these:\n",
            "  64750538 (100.00%) were unpaired; of these:\n",
            "    53995376 (83.39%) aligned 0 times\n",
            "    2941610 (4.54%) aligned exactly 1 time\n",
            "    7813552 (12.07%) aligned >1 times\n",
            "16.61% overall alignment rate\n",
            "CPU times: user 26.6 s, sys: 3.28 s, total: 29.9 s\n",
            "Wall time: 56min 15s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Отбор уникальных ридов"
      ],
      "metadata": {
        "id": "aGeOOz6C-o-X"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -h -S -b -o bowtie2_res/res_ZFO.bam bowtie2_res/res_ZFO.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_JYK.bam bowtie2_res/res_JYK.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_SVH.bam bowtie2_res/res_SVH.sam"
      ],
      "metadata": {
        "id": "rIAo8ByR83DX"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!apt install ldc"
      ],
      "metadata": {
        "id": "NgcpeZ2q7nEE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!git clone --recursive https://github.com/lomereiter/sambamba.git"
      ],
      "metadata": {
        "id": "8n6vI6SK7ukl"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cd sambamba"
      ],
      "metadata": {
        "id": "otgtepgH70XE"
      },
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d sambamba-0.8.2-linux-amd64-static.gz"
      ],
      "metadata": {
        "id": "EPV_1ybI8B5Q"
      },
      "execution_count": 21,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!chmod a+x sambamba-0.8.2-linux-amd64-static"
      ],
      "metadata": {
        "id": "V4FHaqYL9L2E"
      },
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!./sambamba-0.8.2-linux-amd64-static --version"
      ],
      "metadata": {
        "id": "1LVc8wrp9XW7"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!./sambamba-0.8.2-linux-amd64-static sort -t 2 -o bowtie2_res/res_ZFO_s.bam bowtie2_res/res_ZFO.bam\n",
        "!./sambamba-0.8.2-linux-amd64-static sort -t 2 -o bowtie2_res/res_JYK_s.bam bowtie2_res/res_JYK.bam\n",
        "!./sambamba-0.8.2-linux-amd64-static sort -t 2 -o bowtie2_res/res_SVH_s.bam bowtie2_res/res_SVH.bam"
      ],
      "metadata": {
        "id": "wlMD_NHI9TPf",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1d08cd51-3019-4584-d960-7657a5fc9886"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n",
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n",
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./sambamba-0.8.2-linux-amd64-static view -h -t 2 -f bam -F '[XS] == null and not unmapped  and not duplicate' bowtie2_res/res_ZFO_s.bam > bowtie2_res/res_ZFO_s1.bam\n",
        "!./sambamba-0.8.2-linux-amd64-static view -h -t 2 -f bam -F '[XS] == null and not unmapped  and not duplicate' bowtie2_res/res_JYK_s.bam > bowtie2_res/res_JYK_s1.bam\n",
        "!./sambamba-0.8.2-linux-amd64-static view -h -t 2 -f bam -F '[XS] == null and not unmapped  and not duplicate' bowtie2_res/res_SVH_s.bam > bowtie2_res/res_SVH_s1.bam"
      ],
      "metadata": {
        "id": "cLF41bnB9tM_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "c054fe4a-0d83-44f0-9ef4-50db6ac421dd"
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n",
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n",
            "\n",
            "sambamba 0.8.2\n",
            " by Artem Tarasov and Pjotr Prins (C) 2012-2021\n",
            "    LDC 1.27.1 / DMD v2.097.2 / LLVM11.0.0 / bootstrap LDC - the LLVM D compiler (1.27.1)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2_ZFO"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 38,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "bd4dd94f-113d-4313-b769-9d93ef872419"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 09 Mar 2022 23:55:23: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_ZFO_s1.bam -c bowtie2_res/res_SVH_s1.bam -f BAM --outdir macs2_ZFO\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_ZFO_s1.bam']\n",
            "# control file = ['bowtie2_res/res_SVH_s1.bam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:23: #1 read tag files... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:23: #1 read treatment tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:25:  1000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:27:  2000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:30:  3000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:32:  4000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:33: 4339598 reads have been read. \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:33: #1.2 read input tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:36:  1000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:38:  2000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: 2941610 reads have been read. \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 tag size is determined as 51 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 tag size = 51.0 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  total tags in treatment: 4339598 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  tags after filtering in treatment: 3575665 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  Redundant rate of treatment: 0.18 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  total tags in control: 2941610 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  tags after filtering in control: 2501702 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1  Redundant rate of control: 0.15 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #1 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #2 Build Peak Model... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:40: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #2 number of paired peaks: 53441 \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: start model_add_line... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: start X-correlation... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: end of X-cor \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #2 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #2 alternative fragment length(s) may be 50,118,156,571 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #2.2 Generate R script for model : macs2_ZFO/NA_model.r \n",
            "WARNING @ Wed, 09 Mar 2022 23:55:42: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 09 Mar 2022 23:55:42: #2 You may need to consider one of the other alternative d(s): 50,118,156,571 \n",
            "WARNING @ Wed, 09 Mar 2022 23:55:42: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #3 Call peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:42: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 09 Mar 2022 23:55:56: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:06: #4 Write output xls file... macs2_ZFO/NA_peaks.xls \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:06: #4 Write broad peak in broadPeak format file... macs2_ZFO/NA_peaks.broadPeak \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:06: #4 Write broad peak in bed12/gappedPeak format file... macs2_ZFO/NA_peaks.gappedPeak \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:06: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_ZFO_s1.bam \\\n",
        "    -c bowtie2_res/res_SVH_s1.bam \\\n",
        " \t  -f BAM \\\n",
        "\t  --outdir macs2_ZFO"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir macs2_JYK"
      ],
      "metadata": {
        "id": "q5IiUY9ZTbYj"
      },
      "execution_count": 33,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_JYK_s1.bam \\\n",
        "    -c bowtie2_res/res_SVH_s1.bam \\\n",
        " \t  -f BAM \\\n",
        "\t  --outdir macs2_JYK"
      ],
      "metadata": {
        "id": "4ouYSid79SL_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5aaa9f60-5299-4ed7-b4c0-06deecb2c9ba"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 09 Mar 2022 23:56:48: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_JYK_s1.bam -c bowtie2_res/res_SVH_s1.bam -f BAM --outdir macs2_JYK\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_JYK_s1.bam']\n",
            "# control file = ['bowtie2_res/res_SVH_s1.bam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:48: #1 read tag files... \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:48: #1 read treatment tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:51:  1000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:53:  2000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:56:  3000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:56: 3340751 reads have been read. \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:56: #1.2 read input tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:56:59:  1000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:01:  2000000 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: 2941610 reads have been read. \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 tag size is determined as 51 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 tag size = 51.0 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1  total tags in treatment: 3340751 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1  tags after filtering in treatment: 2764856 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1  Redundant rate of treatment: 0.17 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1  total tags in control: 2941610 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:03: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:04: #1  tags after filtering in control: 2501702 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:04: #1  Redundant rate of control: 0.15 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:04: #1 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:04: #2 Build Peak Model... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:04: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #2 number of paired peaks: 41495 \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: start model_add_line... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: start X-correlation... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: end of X-cor \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #2 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #2 alternative fragment length(s) may be 50,120,165,505 bps \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #2.2 Generate R script for model : macs2_JYK/NA_model.r \n",
            "WARNING @ Wed, 09 Mar 2022 23:57:05: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 09 Mar 2022 23:57:05: #2 You may need to consider one of the other alternative d(s): 50,120,165,505 \n",
            "WARNING @ Wed, 09 Mar 2022 23:57:05: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #3 Call peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:05: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:24: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:34: #4 Write output xls file... macs2_JYK/NA_peaks.xls \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:34: #4 Write broad peak in broadPeak format file... macs2_JYK/NA_peaks.broadPeak \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:34: #4 Write broad peak in bed12/gappedPeak format file... macs2_JYK/NA_peaks.gappedPeak \n",
            "INFO  @ Wed, 09 Mar 2022 23:57:34: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 35,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "79593864-67ea-482c-cd9a-0227b3bf7b3e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 23:53:00--  https://www.encodeproject.org/files/ENCFF695GDN/@@download/ENCFF695GDN.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/3660c7f2-fe17-480a-a00b-f5d3321da317/ENCFF695GDN.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF695GDN.bed.gz&AWSAccessKeyId=ASIATGZNGCNXWSMYPUP7&Signature=Ns1963v2Msc278cKmaS7hOehYTs%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHgaCXVzLXdlc3QtMiJIMEYCIQDw1bKycy5XVUrjtNic93jJfP1LDxh5FRianAyG7o9VJwIhAOVFAk82m%2BFnvXPxbBjlSGCWYkEHezK%2FWtHYnKdjeU6VKoMECOH%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgww%2FYT25PJi5pxoilcq1wOj0TcsjOON5OpAwXqcK5Cex8mz9qutvuWAzNz%2FRj5d%2BPIeyywEcfWPiLtVz7cDmZb7T8gq097Vh%2FOByNei7PsZm1r2IiLneJI2ihlg%2BrXUe34tNfsKYgijvoyLsKJU%2FT98fcAnS%2FYoENgfIGFO8R5aDz4mSgpwQ2AZFUeBVORgaqXksmLs%2BB3V13Zup1MhDCDFx3KzAMG%2BDbCnII9UWIvh50R2dSr8Ls3pu1I6IyZebsnIc5aW6iZxH5cgeSAL7RbNRndCrygh9s4ItHrkqtD8wb2RhJMcAQZW6CtZKxzV%2F4sdInmhN6rdtxqvg5xozrZvZM4QKOO6aa4F89Su22UA90sfBVXrtzY77cSDP8%2B5xRPPPaYzGtzO9y0iCvGJEOCe3ZOBEZXTzeFjS6VZNVu0J7%2BX1QVyti5wUvfOG2GGCCxH3lhqzl04mOw2iTx6cfxTR1l5qkgWEXG5lQwdQInhXFI%2FKwPFxMBb2RwybvtVt3%2BFWEPb4aMyir1vHjp18qcMkghSg%2BS3ZeLGW9MGVfIkvZS1xTXiY%2FPTppVGp2z%2FBPXN71rcwytmvYEcXEvTWnAPxhUnO%2Ba2Dxp3cpVuwLa7%2BbrECqz6MsxJZdsMzUZe%2Bsg8flZ6lJww9%2FGkkQY6pAFfi5x5yIDyhsLy16%2BjnbGc%2F5U2K8RRA2CIYvYw3m3n0pve8o6AJfQN0Jv7j4ysqsa1J2fl9Zk3FyrMPLd6xrjBroTxhPEJ9YUraS5y%2BREEzL%2BCYTHzQsNKq23VkUqtAjESVijUm3fpUoznwteNtASUiPh2ZrDuoFlyaDt03Z6gqjNUWKPclqzkt62b6MKLseC6gpXCEptnPRnfBTVCXZIF5FVFEQ%3D%3D&Expires=1646999580 [following]\n",
            "--2022-03-09 23:53:00--  https://encode-public.s3.amazonaws.com/2017/10/13/3660c7f2-fe17-480a-a00b-f5d3321da317/ENCFF695GDN.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF695GDN.bed.gz&AWSAccessKeyId=ASIATGZNGCNXWSMYPUP7&Signature=Ns1963v2Msc278cKmaS7hOehYTs%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHgaCXVzLXdlc3QtMiJIMEYCIQDw1bKycy5XVUrjtNic93jJfP1LDxh5FRianAyG7o9VJwIhAOVFAk82m%2BFnvXPxbBjlSGCWYkEHezK%2FWtHYnKdjeU6VKoMECOH%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgww%2FYT25PJi5pxoilcq1wOj0TcsjOON5OpAwXqcK5Cex8mz9qutvuWAzNz%2FRj5d%2BPIeyywEcfWPiLtVz7cDmZb7T8gq097Vh%2FOByNei7PsZm1r2IiLneJI2ihlg%2BrXUe34tNfsKYgijvoyLsKJU%2FT98fcAnS%2FYoENgfIGFO8R5aDz4mSgpwQ2AZFUeBVORgaqXksmLs%2BB3V13Zup1MhDCDFx3KzAMG%2BDbCnII9UWIvh50R2dSr8Ls3pu1I6IyZebsnIc5aW6iZxH5cgeSAL7RbNRndCrygh9s4ItHrkqtD8wb2RhJMcAQZW6CtZKxzV%2F4sdInmhN6rdtxqvg5xozrZvZM4QKOO6aa4F89Su22UA90sfBVXrtzY77cSDP8%2B5xRPPPaYzGtzO9y0iCvGJEOCe3ZOBEZXTzeFjS6VZNVu0J7%2BX1QVyti5wUvfOG2GGCCxH3lhqzl04mOw2iTx6cfxTR1l5qkgWEXG5lQwdQInhXFI%2FKwPFxMBb2RwybvtVt3%2BFWEPb4aMyir1vHjp18qcMkghSg%2BS3ZeLGW9MGVfIkvZS1xTXiY%2FPTppVGp2z%2FBPXN71rcwytmvYEcXEvTWnAPxhUnO%2Ba2Dxp3cpVuwLa7%2BbrECqz6MsxJZdsMzUZe%2Bsg8flZ6lJww9%2FGkkQY6pAFfi5x5yIDyhsLy16%2BjnbGc%2F5U2K8RRA2CIYvYw3m3n0pve8o6AJfQN0Jv7j4ysqsa1J2fl9Zk3FyrMPLd6xrjBroTxhPEJ9YUraS5y%2BREEzL%2BCYTHzQsNKq23VkUqtAjESVijUm3fpUoznwteNtASUiPh2ZrDuoFlyaDt03Z6gqjNUWKPclqzkt62b6MKLseC6gpXCEptnPRnfBTVCXZIF5FVFEQ%3D%3D&Expires=1646999580\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.228.51\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.228.51|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1953612 (1.9M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF695GDN.bed.gz’\n",
            "\n",
            "ENCFF695GDN.bed.gz  100%[===================>]   1.86M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2022-03-09 23:53:00 (18.8 MB/s) - ‘ENCFF695GDN.bed.gz’ saved [1953612/1953612]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF695GDN/@@download/ENCFF695GDN.bed.gz\n",
        "!gzip -d ENCFF695GDN.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 41,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6b34ac70-5644-45c4-ea4a-2912e1b0f35a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2_ZFO/NA_peaks.broadPeak ENCFF695GDN.bed --filenames --output venn_results/venn1.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 43,
      "metadata": {
        "id": "7qqn3n2cRdkt",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4ae8a75f-e00e-4cd7-fb8c-d196198e13f9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF695GDN.bed macs2_ZFO/NA_peaks.broadPeak --filenames --output venn_results/venn2.jpg"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2_JYK/NA_peaks.broadPeak ENCFF695GDN.bed --filenames --output venn_results/venn3.jpg\n",
        "!intervene venn -i ENCFF695GDN.bed macs2_JYK/NA_peaks.broadPeak --filenames --output venn_results/venn4.jpg"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dy_ngUoUTR5m",
        "outputId": "1506e6d3-ee84-4233-aaa1-f3821570dcbe"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn3.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn4.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [
        "-z4GQiMfU1wf",
        "6sPcIefNHx-L"
      ],
      "name": "ChIP-seq.ipynb",
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}