{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/dabardonov/hse_hw2_chip/blob/main/ChIP_seq.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "lrtRbP3fAvj8"
      },
      "outputs": [],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XnbIt77YBOT9"
      },
      "outputs": [],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "35a5573a-d3af-49ab-909c-806f64a9f967"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00,  6.06it/s]\n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:17<00:00, 17.94s/it]\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00, 13.04it/s]\n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "F0BLWsH2n3Ov"
      },
      "outputs": [],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Ce6hiSTdkarU"
      },
      "outputs": [],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "AUML3dvsqT9E"
      },
      "outputs": [],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "d88b978b-313d-475d-9d8f-0003c9b40cc9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-08 12:19:06--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  3.29MB/s    in 3.0s    \n",
            "\n",
            "2022-03-08 12:19:10 (3.29 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "f1d5a213-85d4-411b-9ebd-37a3dbb68c0b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 11:22:27--  https://www.encodeproject.org/files/ENCFF195ZFO/@@download/ENCFF195ZFO.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/04a0a840-ff6d-41b1-8ec7-a11d97fe6627/ENCFF195ZFO.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF195ZFO.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=2QRqw73vL0krg2jhfj0ewPuyEN8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954548 [following]\n",
            "--2022-03-09 11:22:28--  https://encode-public.s3.amazonaws.com/2017/10/13/04a0a840-ff6d-41b1-8ec7-a11d97fe6627/ENCFF195ZFO.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF195ZFO.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=2QRqw73vL0krg2jhfj0ewPuyEN8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954548\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.194.121\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.194.121|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3381896063 (3.1G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF195ZFO.fastq.gz’\n",
            "\n",
            "ENCFF195ZFO.fastq.g 100%[===================>]   3.15G  33.1MB/s    in 99s     \n",
            "\n",
            "2022-03-09 11:24:08 (32.5 MB/s) - ‘ENCFF195ZFO.fastq.gz’ saved [3381896063/3381896063]\n",
            "\n",
            "--2022-03-09 11:24:08--  https://www.encodeproject.org/files/ENCFF734JYK/@@download/ENCFF734JYK.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/415f0016-d8c2-4bd7-a437-7b9baa51c2af/ENCFF734JYK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF734JYK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=BLVMH4GsHovCk7EBvRMinMci%2Fro%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954648 [following]\n",
            "--2022-03-09 11:24:08--  https://encode-public.s3.amazonaws.com/2017/10/13/415f0016-d8c2-4bd7-a437-7b9baa51c2af/ENCFF734JYK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF734JYK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=BLVMH4GsHovCk7EBvRMinMci%2Fro%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954648\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.253.59\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.253.59|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2723479828 (2.5G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF734JYK.fastq.gz’\n",
            "\n",
            "ENCFF734JYK.fastq.g 100%[===================>]   2.54G  34.0MB/s    in 78s     \n",
            "\n",
            "2022-03-09 11:25:27 (33.1 MB/s) - ‘ENCFF734JYK.fastq.gz’ saved [2723479828/2723479828]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF195ZFO/@@download/ENCFF195ZFO.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF734JYK/@@download/ENCFF734JYK.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF151SVH/@@download/ENCFF151SVH.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NxVWhYSbfshg",
        "outputId": "0e3de900-8407-4679-85d6-c68cfac06ecd"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 11:25:27--  https://www.encodeproject.org/files/ENCFF151SVH/@@download/ENCFF151SVH.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/bffb72a4-28e5-4f77-98b9-358feb1fd186/ENCFF151SVH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151SVH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=eIwrqqC%2B6oDXCGCXbWjGOu8fG8Y%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954728 [following]\n",
            "--2022-03-09 11:25:28--  https://encode-public.s3.amazonaws.com/2017/10/13/bffb72a4-28e5-4f77-98b9-358feb1fd186/ENCFF151SVH.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151SVH.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXUH7D5Z5G&Signature=eIwrqqC%2B6oDXCGCXbWjGOu8fG8Y%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGsaCXVzLXdlc3QtMiJHMEUCIQC1BhCTaFJJw3qXnOLjRMyL%2F65ze57kiw3zjeaq2O9EDwIgZNufJ3wTAbY50irgUnAapCPzkCH2h3XNOGScswa4pAsqgwQI0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDCrFE01%2FPzBTLSjefirXA2CSGwGEt960XqMeKTKlNC%2BRFDcF9xh%2F22ijfcXSGTJcVmGYuxFdKFo2QnmTk0Nr5dK3dpppUToROt0TrrgTjq7GudGBO1hWUbzg5DwtmD7u52xROUXGPQEjTC0ZMLrnQ7vnCEawvv4aCsyd3rfgqRNj2SjeYI4KgQljtcjfbeMZGh1nJeEf30%2BOKoJOHN33xCzCLkVW7b5Ij1cSyQzGhg72hboEtFptWLM5ekEK4feD9P0BZsKi8syUmG%2BSA9ANUwRPzwqCAEaG5s36UpTqVQu8YeWAghYNAz53Br%2FUF7wsOTGPRmC9pEDcl%2BIAvm3jQd9WLJcuQb5yWT%2BNNrlqSTFm1ubKGKE6mqrjojlA2WPhY4%2BvIJvbKCmqA9Ofd6oIM90mTgjymcs3MFppHtiivX3KkPX41wUmNbKg22OK74v7xpdYBBbfGcJGBovK%2FkCaTrAf%2FX6BP%2BfwedCAdm5Kv7Gi8ogLc829ejG2ErBU4Vto1U8R09F0Fuhq6%2FDFlTrXsyXnPLtJ3qmVZN6m6aMHh0ikKa%2Fun8w1CXsUWRuujCQ42bccY3vKWjh9hfepDCMsVonRA%2BWNWkFFRflLCHx669gWuMGfG%2B6z1heyYimviKn5ORrqgzYW7DDq%2FqGRBjqlAZDZEr8pn%2FLHV0SxcgSrX08tYZYDCItqtL4KCzAcrz58M11TlgaOvYJvUnugp5L3Ih6udr1XiVfDHWzf9Umh2WQJbl5Jn1ZXkp4NYJ5ucjw3JCyc%2FAgM12RtIrSogt5IMYZvN20lx71URTjY3Cda4gtsTK3w0YUXclta%2BcviU7%2BK2JjveyJw%2FsvWBSuXp540K%2F20FpbBAVxC8gYzuX8LG5wl1iKhqw%3D%3D&Expires=1646954728\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.251.83\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.251.83|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1974036260 (1.8G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF151SVH.fastq.gz’\n",
            "\n",
            "ENCFF151SVH.fastq.g 100%[===================>]   1.84G  32.0MB/s    in 59s     \n",
            "\n",
            "2022-03-09 11:26:27 (31.8 MB/s) - ‘ENCFF151SVH.fastq.gz’ saved [1974036260/1974036260]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "!gzip -d /content/ENCFF195ZFO.fastq.gz /content/ENCFF734JYK.fastq.gz /content/ENCFF151SVH.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5XM6jBcuGjMV"
      },
      "outputs": [],
      "source": [
        "!./FastQC/fastqc /content/ENCFF195ZFO.fastq\n",
        "!./FastQC/fastqc /content/ENCFF734JYK.fastq\n",
        "!./FastQC/fastqc /content/ENCFF151SVH.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "Подрезать чтения не нужно"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "N4JIGZOsGiY2"
      },
      "outputs": [],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
        "!gzip -d chr14.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "bhraQ6NdGiPN"
      },
      "outputs": [],
      "source": [
        "%%time\n",
        "!bowtie2-build chr14.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4abd5139-a4db-4871-d83e-35f3ab8ca561"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "109131346 reads; of these:\n",
            "  109131346 (100.00%) were unpaired; of these:\n",
            "    94077115 (86.21%) aligned 0 times\n",
            "    4339598 (3.98%) aligned exactly 1 time\n",
            "    10714633 (9.82%) aligned >1 times\n",
            "13.79% overall alignment rate\n",
            "CPU times: user 35 s, sys: 5.12 s, total: 40.1 s\n",
            "Wall time: 1h 22min 7s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF195ZFO.fastq \\\n",
        "          -S bowtie2_res/res_ZFO.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ad8a8326-c319-4ae4-8485-5b807ce737df"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "88120815 reads; of these:\n",
            "  88120815 (100.00%) were unpaired; of these:\n",
            "    76904496 (87.27%) aligned 0 times\n",
            "    3340751 (3.79%) aligned exactly 1 time\n",
            "    7875568 (8.94%) aligned >1 times\n",
            "12.73% overall alignment rate\n",
            "CPU times: user 28.2 s, sys: 3.97 s, total: 32.2 s\n",
            "Wall time: 1h 5min 29s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF734JYK.fastq \\\n",
        "          -S bowtie2_res/res_JYK.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF151SVH.fastq \\\n",
        "          -S bowtie2_res/res_SVH.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Sc4zGUdUp9eP",
        "outputId": "161596bb-b244-4939-a1ff-9c15413b715b"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "64750538 reads; of these:\n",
            "  64750538 (100.00%) were unpaired; of these:\n",
            "    53995376 (83.39%) aligned 0 times\n",
            "    2941610 (4.54%) aligned exactly 1 time\n",
            "    7813552 (12.07%) aligned >1 times\n",
            "16.61% overall alignment rate\n",
            "CPU times: user 22.9 s, sys: 3.48 s, total: 26.4 s\n",
            "Wall time: 53min 46s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EJQOCcv2nJW_"
      },
      "outputs": [],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_ZFO.sam \\\n",
        "    -c bowtie2_res/res_SVH.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir macs2_JYK"
      ],
      "metadata": {
        "id": "q5IiUY9ZTbYj"
      },
      "execution_count": 14,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_JYK.sam \\\n",
        "    -c bowtie2_res/res_SVH.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2_JYK"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4ouYSid79SL_",
        "outputId": "77159336-116e-4012-aafb-da5f1fb6973f"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 09 Mar 2022 15:59:32: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_JYK.sam -c bowtie2_res/res_SVH.sam -f SAM --outdir macs2_JYK\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_JYK.sam']\n",
            "# control file = ['bowtie2_res/res_SVH.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:32: #1 read tag files... \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:32: #1 read treatment tags... \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:45:  12000000 \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:51:  17000000 \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:54:  20000000 \n",
            "INFO  @ Wed, 09 Mar 2022 15:59:57:  23000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:00:10:  34000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:00:26:  47000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:00:27:  48000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:00:52:  71000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:01:  80000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:04:  83000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:05:  84000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:11: #1.2 read input tags... \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:18:  6000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:19:  7000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:25:  12000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:28:  15000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:29:  16000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:39:  25000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:43:  28000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:45:  30000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:49:  33000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:51:  35000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:01:54:  38000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:12:  54000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:18:  59000000 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 tag size is determined as 51 bps \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 tag size = 51.0 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  total tags in treatment: 11216319 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  tags after filtering in treatment: 7560421 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  Redundant rate of treatment: 0.33 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  total tags in control: 10755162 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  tags after filtering in control: 7677284 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1  Redundant rate of control: 0.29 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #1 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #2 Build Peak Model... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:26: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:27: #2 number of paired peaks: 33503 \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:27: start model_add_line... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: start X-correlation... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: end of X-cor \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #2 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #2 predicted fragment length is 51 bps \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #2 alternative fragment length(s) may be 51 bps \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #2.2 Generate R script for model : macs2_JYK/NA_model.r \n",
            "WARNING @ Wed, 09 Mar 2022 16:02:28: #2 Since the d (51) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 09 Mar 2022 16:02:28: #2 You may need to consider one of the other alternative d(s): 51 \n",
            "WARNING @ Wed, 09 Mar 2022 16:02:28: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #3 Call peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 09 Mar 2022 16:02:28: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 09 Mar 2022 16:03:05: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 09 Mar 2022 16:03:35: #4 Write output xls file... macs2_JYK/NA_peaks.xls \n",
            "INFO  @ Wed, 09 Mar 2022 16:03:35: #4 Write broad peak in broadPeak format file... macs2_JYK/NA_peaks.broadPeak \n",
            "INFO  @ Wed, 09 Mar 2022 16:03:35: #4 Write broad peak in bed12/gappedPeak format file... macs2_JYK/NA_peaks.gappedPeak \n",
            "INFO  @ Wed, 09 Mar 2022 16:03:35: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "781354e2-3a5b-427e-f555-4e50f9ade2ba"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 16:06:08--  https://www.encodeproject.org/files/ENCFF695GDN/@@download/ENCFF695GDN.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/10/13/3660c7f2-fe17-480a-a00b-f5d3321da317/ENCFF695GDN.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF695GDN.bed.gz&AWSAccessKeyId=ASIATGZNGCNX2SMNLPK3&Signature=bfHuNvCiej1ALKEtkItBK%2FfEtP0%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHAaCXVzLXdlc3QtMiJIMEYCIQCJMuQ108GWYJwAmDaWb5wJRnS2Uy7F0yi3RxTgcYrEjwIhAMZGvJWkRLugcSB7o3304WBAc3wafQPb89OSLL8YZHHCKoMECNn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzcR1CeoV%2Bc%2BthG9A8q1wOozo9C3H%2FZ%2FhQJ5xbvJ%2Bo%2BOveEV87IavnwzBkmjGi2JiTdDhZ8V2h3VQxAmHnZkQmQZk15i0N5QBFZUVm8OSo3Xl%2BE9hqf0cZqY3E%2FZpOs%2B4xeKfeaM2BnovOMqVAa9%2F0gYMBR3Fi5v35nxV9PjJ9%2B38%2FyE4GJZdhGXRTdI6xyOaSq2LmylBUr9an7k4sn0UhCwwfmE0FBrQH2ILJPaNu%2B78syOMZM8p0rCXyYnWFQu6OfC9J4vtztSsG7PU5dn5KaxoNtiwj6XKovFT46ubcpH5XWTIKXGids2EDAXFZ5%2FkRnIiSjqOFtO0waXqoP%2F%2B0AC8RjMjRXAXOtPEvE%2FEV5wa9sAeFHxCCUoxV1sW4t5US3D%2F6MlLxNAXQLs8OA6V5Ge5WdMqhfRPKkvIIqOjTF%2FQ%2FA60DlTrDOwDAo5GoWZ1jh%2BtyLC4GgHybmrCExq%2F1TR%2FeGPV5xRmGBggGlnsADesTlq2U3c%2BNuKPJI9PL1kzmQI3kyJjDu3CreN3pZ8sCXhjGWdNgiIbSg5uPXhlRVFKJF1A%2FcJcTaoCcx3OzHj%2BNanFa5Qh5OOR2HNfLdesfjz6fiEO6EnU0Nwdi3AvKKyoSldOr9vg9eI34leEYLg%2B0GPvI4Yasw%2Fo2jkQY6pAHRY0kBJ3XKbSWUd%2FGvQNl%2F8EUv%2BWc5lYIIhpgYK%2FixOl%2FDMiu7MBCifHzaLvToQtI9bWtQwduy2c6VIm1sqlGLFkawjR3cxWhlWpHbi8Tb913mEojjH5XMge3Bw9efUC6ImeYKQkLzJKLi74T4sR0coPcRj%2FvhVsa%2FT7uqIpxo4CqxWYcr7%2FZpbdcikZQ4gOjqx6414Bp4QKj8ets4eUZNTfrqjQ%3D%3D&Expires=1646971568 [following]\n",
            "--2022-03-09 16:06:08--  https://encode-public.s3.amazonaws.com/2017/10/13/3660c7f2-fe17-480a-a00b-f5d3321da317/ENCFF695GDN.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF695GDN.bed.gz&AWSAccessKeyId=ASIATGZNGCNX2SMNLPK3&Signature=bfHuNvCiej1ALKEtkItBK%2FfEtP0%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHAaCXVzLXdlc3QtMiJIMEYCIQCJMuQ108GWYJwAmDaWb5wJRnS2Uy7F0yi3RxTgcYrEjwIhAMZGvJWkRLugcSB7o3304WBAc3wafQPb89OSLL8YZHHCKoMECNn%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzcR1CeoV%2Bc%2BthG9A8q1wOozo9C3H%2FZ%2FhQJ5xbvJ%2Bo%2BOveEV87IavnwzBkmjGi2JiTdDhZ8V2h3VQxAmHnZkQmQZk15i0N5QBFZUVm8OSo3Xl%2BE9hqf0cZqY3E%2FZpOs%2B4xeKfeaM2BnovOMqVAa9%2F0gYMBR3Fi5v35nxV9PjJ9%2B38%2FyE4GJZdhGXRTdI6xyOaSq2LmylBUr9an7k4sn0UhCwwfmE0FBrQH2ILJPaNu%2B78syOMZM8p0rCXyYnWFQu6OfC9J4vtztSsG7PU5dn5KaxoNtiwj6XKovFT46ubcpH5XWTIKXGids2EDAXFZ5%2FkRnIiSjqOFtO0waXqoP%2F%2B0AC8RjMjRXAXOtPEvE%2FEV5wa9sAeFHxCCUoxV1sW4t5US3D%2F6MlLxNAXQLs8OA6V5Ge5WdMqhfRPKkvIIqOjTF%2FQ%2FA60DlTrDOwDAo5GoWZ1jh%2BtyLC4GgHybmrCExq%2F1TR%2FeGPV5xRmGBggGlnsADesTlq2U3c%2BNuKPJI9PL1kzmQI3kyJjDu3CreN3pZ8sCXhjGWdNgiIbSg5uPXhlRVFKJF1A%2FcJcTaoCcx3OzHj%2BNanFa5Qh5OOR2HNfLdesfjz6fiEO6EnU0Nwdi3AvKKyoSldOr9vg9eI34leEYLg%2B0GPvI4Yasw%2Fo2jkQY6pAHRY0kBJ3XKbSWUd%2FGvQNl%2F8EUv%2BWc5lYIIhpgYK%2FixOl%2FDMiu7MBCifHzaLvToQtI9bWtQwduy2c6VIm1sqlGLFkawjR3cxWhlWpHbi8Tb913mEojjH5XMge3Bw9efUC6ImeYKQkLzJKLi74T4sR0coPcRj%2FvhVsa%2FT7uqIpxo4CqxWYcr7%2FZpbdcikZQ4gOjqx6414Bp4QKj8ets4eUZNTfrqjQ%3D%3D&Expires=1646971568\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.218.251\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.218.251|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1953612 (1.9M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF695GDN.bed.gz’\n",
            "\n",
            "ENCFF695GDN.bed.gz  100%[===================>]   1.86M  3.52MB/s    in 0.5s    \n",
            "\n",
            "2022-03-09 16:06:09 (3.52 MB/s) - ‘ENCFF695GDN.bed.gz’ saved [1953612/1953612]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF695GDN/@@download/ENCFF695GDN.bed.gz\n",
        "!gzip -d ENCFF695GDN.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "31912261-b46b-453a-e316-f6720abaa57d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/NA_peaks.broadPeak ENCFF695GDN.bed --filenames --output venn_results/venn1.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "id": "7qqn3n2cRdkt",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9671cc4f-008f-40de-f608-1f312bd6f1de"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF695GDN.bed macs2/NA_peaks.broadPeak --filenames --output venn_results/venn2.jpg"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2_JYK/NA_peaks.broadPeak ENCFF695GDN.bed --filenames --output venn_results/venn3.jpg\n",
        "!intervene venn -i ENCFF695GDN.bed macs2_JYK/NA_peaks.broadPeak --filenames --output venn_results/venn4.jpg"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dy_ngUoUTR5m",
        "outputId": "08b13214-db2b-44ce-c231-a44f9b5de927"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn3.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn4.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [
        "XUZS4GUkHIEG",
        "vcr5CjibHTgF",
        "-z4GQiMfU1wf",
        "6sPcIefNHx-L"
      ],
      "name": "ChIP-seq.ipynb",
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}